MPF.Core/SubmissionInfoTool.cs

```diff
@@ -670,6 +671,65 @@ public async static Task<bool> FillFromRedump(Options options, SubmissionInfo in
             return true;
         }
 
-#endregion
+        #endregion
+
+        #region Helper Functions
+
+        /// <summary>
+        /// Formats a list of volume labels and their corresponding filesystems
+        /// </summary>
+        /// <param name="labels">Dictionary of volume labels and their filesystems</param>
+        /// <returns>Formatted string of volume labels and their filesystems</returns>
+        private static string? FormatVolumeLabels(string? driveLabel, Dictionary<string, List<string>>? labels)
+        {
+
+            // Must have at least one label to format
+            if (driveLabel == null && (labels == null || labels.Count <= 0))
+                return null;
+
+            // If no labels given, use drive label
+            if (labels == null || labels.Count <= 0)
+            {
+                // Ignore common volume labels
+                if (Drive.GetRedumpSystemFromVolumeLabel(driveLabel) != null)
+                    return null;
+
+                return driveLabel;
+            }
+
+            // If only one label, don't mention fs
+            string firstLabel = labels.FirstOrDefault().Key;
```

Comment:
Since you know that the labels contains at least one item at this point, I think you can get away with just `.First`